lib/dgram.js

```diff
@@ -336,7 +336,10 @@ function afterSend(err) {
   if (err) {
     err = exceptionWithHostPort(err, 'send', this.address, this.port);
   }
-  this.callback(err, this.length);
+  var self = this;
+  setImmediate(function() {
+    self.callback(err, self.length);
+  });
```

Comment:
I tried earlier today.  setImmediate() works but process.nextTick() still hangs.  From some ad hoc testing I got the impression that the event loop is making some progress - i.e. it's not that dgram.send() is truly synchronous - but it's not making enough progress for the setTimeout timer in benchmark/net/dgram.js to fire.